Automattic/wp-calypso
# Add state infrastructure for marketplace flow state

#### Changes proposed in this Pull Request
* Sets up a slice of redux state to be use to determine flow logic across the purchase flow. The state itself is evolving and currently we will at least need the primaryDomain information to make decisions across the stages of the flow and so it has been added as an initial data point.

#### Testing instructions
* Make sure you have redux tools installed
* Go to the marketplace domains page with the feature flag enabled /plugins/domain/{siteUrl}?flags=marketplace-yoast
     * Eg: http://calypso.localhost:3000/plugins/domain/awesome.com?flags=marketplace-yoast
* Click on a domain  suggestion
* Make sure the below action is fired and the state update is correct
![image](https://user-images.githubusercontent.com/3422709/118928179-7f611580-b960-11eb-9ae7-81ec27d28c0f.png)


Related to #52081
